File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    queue = [];
    const array = s.split(' ');

    for (let i = 0; i < array.length; i++ ){
        if(array[i].length){
            queue.unshift(array[i]); 
        };
    };

    return queue.join(' ')
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        arr = s.split()
        l = 0
        r = len(arr) - 1

        while l < r:
            arr[l], arr[r] = arr[r], arr[l]
            l += 1
            r -= 1
        
        return " ".join(arr)